migration/2.1.3/SequencerInbox.sol

```diff
@@ -0,0 +1,843 @@
+        );
+        // Can only force-include after the Sequencer-only window has expired.
+        if (l1BlockAndTime[0] + delayBlocks >= block.number) revert ForceIncludeBlockTooSoon();
+        if (l1BlockAndTime[1] + delaySeconds >= block.timestamp) revert ForceIncludeTimeTooSoon();
+
+        // Verify that message hash represents the last message sequence of delayed message to be included
+        bytes32 prevDelayedAcc = 0;
+        if (_totalDelayedMessagesRead > 1) {
+            prevDelayedAcc = bridge.delayedInboxAccs(_totalDelayedMessagesRead - 2);
+        }
+        if (
+            bridge.delayedInboxAccs(_totalDelayedMessagesRead - 1) !=
+            Messages.accumulateInboxMessage(prevDelayedAcc, messageHash)
+        ) revert IncorrectMessagePreimage();
+
+        (bytes32 dataHash, IBridge.TimeBounds memory timeBounds) = formEmptyDataHash(
+            _totalDelayedMessagesRead
+        );
+        uint256 __totalDelayedMessagesRead = _totalDelayedMessagesRead;
+        uint256 prevSeqMsgCount = bridge.sequencerReportedSubMessageCount();
+        uint256 newSeqMsgCount = prevSeqMsgCount; // force inclusion should not modify sequencer message count
+        (
+            uint256 seqMessageIndex,
+            bytes32 beforeAcc,
+            bytes32 delayedAcc,
+            bytes32 afterAcc
+        ) = addSequencerL2BatchImpl(
+                dataHash,
+                __totalDelayedMessagesRead,
+                0,
+                prevSeqMsgCount,
+                newSeqMsgCount
+            );
+        emit SequencerBatchDelivered(
+            seqMessageIndex,
+            beforeAcc,
+            afterAcc,
+            delayedAcc,
+            totalDelayedMessagesRead,
+            timeBounds,
+            IBridge.BatchDataLocation.NoData
+        );
+    }
+
+    /// @dev Deprecated, kept for abi generation and will be removed in the future
+    function addSequencerL2BatchFromOrigin(
+        uint256 sequencerNumber,
+        bytes calldata data,
+        uint256 afterDelayedMessagesRead,
+        IGasRefunder gasRefunder
+    ) external pure {
+        revert Deprecated();
+    }
+
+    /**
+        Deprecated because we added a new method with TEE attestation quote
+        to verify that the batch is posted by the batch poster running in TEE.
+     */
+    function addSequencerL2BatchFromOrigin(
+        uint256 sequenceNumber,
+        bytes calldata data,
+        uint256 afterDelayedMessagesRead,
+        IGasRefunder gasRefunder,
+        uint256 prevMessageCount,
+        uint256 newMessageCount
+    ) external refundsGas(gasRefunder, IReader4844(address(0))) {
+        // solhint-disable-next-line avoid-tx-origin
+        if(!CallerChecker.isCallerCodelessOrigin()) revert NotCodelessOrigin();
+        if (msg.sender != tx.origin) revert NotOrigin();
+        if (!isBatchPoster[msg.sender]) revert NotBatchPoster();
+        (bytes32 dataHash, IBridge.TimeBounds memory timeBounds) = formCallDataHash(
```

Comment:
dude am I tripping or is espresso code missing from here?

Reply:
Yeah this is the revert contract that would be used to revert to a non-espresso sequencer inbox.
All we would need to do to revert to the sequencer inbox used before the 2.1.3 upgrade would be deploy an existing migration action pointing to the sequencer inbox implementation that rari is already using.